### PR TITLE
asymmetric lineshapes for eds and xrf fitting

### PR DESCRIPTION
**Pull request by pquinn-dls**
### Description of the change

In fitting XRF peaks an asymmetric peak is often needed to fit an elastic or compton peak.
I have implemented 
1) Doniach lineshape  - this is an asymmetric lineshape originally designed for XPS but
it fits tail like functions quite well
2) Split Voigt - this a voigt but a different sigma on the left and right of the centre

### Progress of the PR
- Doniach implemented as Expression,
- SplitVoigt implemented as Component
- docstrings with latex equations (if appropriate),
- tests implemented

- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
>>> g1 = hs.components1D.Doniach(A=100.0,centre=0.0, sigma=1.0, alpha=0.1)
>>> g2 = hs.components1D.SplitVoigt(A=100.0,centre=0.0, sigma1=1.0,sigma2=2.0,fraction=0.1)

### CODE REVIEW

**Comment by pquinn-dls** [comment]
Tests failing on CI due to tifffile... ?

**Comment by ericpre** [comment]
> Tests failing on CI due to tifffile... ?

Yes, this is fixed in #2323.

xref #2269.

**Comment by pquinn-dls** [comment]
@ericpre - sorry Eric it's not clear what I do next  - wait for the patch #2323 to merge  ?

**Comment by ericpre** [comment]
For the moment, nothing to be done here with respect to the tiff tests. Once #2323 is merged, it may be worth merging the `RELEASE_next_minor` into this branch to get CI green.
In the meantime, this is less convenient than usual, because we need to check that only the tiff tests are failing, instead of relying on the CI tick boxes being green.

File: hyperspy/_components/doniach.py
Comment (francisco-dlp) [inline hyperspy/_components/doniach.py]: This equation doesn't compile properly using sphinx.

File: hyperspy/_components/split_pvoigt.py
Comment (francisco-dlp) [inline hyperspy/_components/split_pvoigt.py]: Same issue as with the other equation.

**Review by francisco-dlp — CHANGES_REQUESTED** [review]
Except for the Latex issues, this looks very good to me. I would only suggest adding the new components to the [list of components in the User Guide](https://hyperspy.org/hyperspy-doc/current/user_guide/model.html#creating-components-for-the-model).

**Comment by francisco-dlp** [comment]
@pquinn-dls, have you noticed https://github.com/pquinn-dls/hyperspy/pull/5?

**Comment by pquinn-dls** [comment]
Sorry @francisco-dlp didn't see this and was working offline - I'll incorporate the changes, Thanks!

**Pull request merged**
